### DISCUSSION TITLE
Get data from the unpublished object.

### DISCUSSION THREAD

**Discussion by inao-cz** [pimcore/pimcore#8052] [Q&A / Support]
Hi,
I'm working on a project that uses Pimcore v6.6.11 and now, I've got to solve task that doesn't seem that easy. When I try to get data from user profile that has been unpublished (not deleted), pimcore returns null. But when I get parent object, data is visible trough Debugger. Is that a bug? What can I do?
Thank you :)

**Comment by dvesh3** [comment]
please share your code snippet to fetch the data? thanks!

**Reply by inao-cz** [reply]
I cannot share snippet but I can tell you what's happening there. I'm getting custom object by it's value from DB. This object has a relation to invoice (another custom object) and this invoice is related to user. When I fetch user from invoice object and try to access Invoice object in twig, I can see that user is loaded. But when I try to use "invoice.user" it becomes null (As seen on these pictures)

![obrazek](https://user-images.githubusercontent.com/29712695/106593244-d9280600-6550-11eb-848b-5f2868e0bf3c.png)
![obrazek](https://user-images.githubusercontent.com/29712695/106593300-f066f380-6550-11eb-8563-08fe5ed09c29.png)

**Reply by dpfaffenbauer** [reply]
that is intentional. If you need that, you have to wrap the call into another call: https://github.com/coreshop/Pimcore/blob/master/DataObject/InheritanceHelper.php#L30

**Reply by inao-cz** [reply]
Solved: Since this code is used on Invoice generating that should be static from a point where the purchase was made (the previous programmer made it fully dynamic - Data can change depending on the time when you will generate the invoice again). I've now used versions of an object. That allows me to get even unpublished data objects.
